pkg/services/ngalert/schedule/registry.go

```diff
@@ -111,3 +111,52 @@ type evaluation struct {
 	scheduledAt time.Time
 	version     int64
 }
+
+type schedulableAlertRulesRegistry struct {
+	rules map[models.AlertRuleKey]*models.SchedulableAlertRule
+	mu    sync.Mutex
+}
```

Comment:
I think eventually we should merge these two registries into one so we will make sure of the consistency.

Reply:
It might be easier to merge them sooner rather than later as I am finding it hard to reason about the scheduler when we have separate registries.